server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DeleteTemplateAction.java

```diff
@@ -76,21 +78,50 @@ private void doHandle(DeleteTemplateWsRequest request) {
         request.getTemplateId(), request.getOrganization(), request.getTemplateName()));
       checkGlobalAdmin(userSession, template.getOrganizationUuid());
 
-      checkTemplateUuidIsNotDefault(template.getUuid());
+      DefaultTemplates defaultTemplates = retrieveDefaultTemplates(dbSession, template);
+
+      checkTemplateUuidIsNotDefault(template, defaultTemplates);
       dbClient.permissionTemplateDao().deleteById(dbSession, template.getId());
+      updateViewDefaultTemplateWhenGovernanceIsNotInstalled(dbSession, template, defaultTemplates);
+
       dbSession.commit();
     }
   }
 
+  /**
+   * The default template for view can be removed when Governance is not installed. To avoid keeping a reference
+   * to a non existing template, we update the default templates.
+   */
+  private void updateViewDefaultTemplateWhenGovernanceIsNotInstalled(DbSession dbSession, PermissionTemplateDto template, DefaultTemplates defaultTemplates) {
+    String viewDefaultTemplateUuid = defaultTemplates.getView();
+    if (viewDefaultTemplateUuid != null && viewDefaultTemplateUuid.equals(template.getUuid())) {
+      defaultTemplates.setView(null);
+      dbClient.organizationDao().setDefaultTemplates(dbSession, template.getOrganizationUuid(), defaultTemplates);
+    }
+  }
+
+  private DefaultTemplates retrieveDefaultTemplates(DbSession dbSession, PermissionTemplateDto template) {
+    return checkFoundWithOptional(
+      dbClient.organizationDao().getDefaultTemplates(dbSession, template.getOrganizationUuid()),
+      "Can't find default templates of Organization with uuid '%s' to which template with uuid '%s' belongs",
+      template.getOrganizationUuid(), template.getUuid());
+  }
+
+  private void checkTemplateUuidIsNotDefault(PermissionTemplateDto template, DefaultTemplates defaultTemplates) {
+    DefaultTemplatesResolverImpl.ResolvedDefaultTemplates resolvedDefaultTemplates = defaultTemplatesResolver.resolve(defaultTemplates);
```

Comment:
`DefaultTemplatesResolverImpl` should be replaced by `DefaultTemplatesResolver`.